pkg/bench/rttanalysis/validate_benchmark_data.go

```diff
@@ -58,193 +60,152 @@ var (
 
 // RunBenchmarkExpectationTests runs tests to validate or rewrite the contents
 // of the benchmark expectations file.
-func RunBenchmarkExpectationTests(t *testing.T) {
-	expectations := readExpectationsFile(t)
-
-	benchmarks := getBenchmarks(t)
-	if *rewriteFlag != "" {
-		rewriteBenchmarkExpectations(t, benchmarks)
+func runBenchmarkExpectationTests(t *testing.T, r *Registry) {
+	if util.IsMetamorphicBuild() {
+		execTestSubprocess(t)
 		return
 	}
 
+	defer log.Scope(t).Close(t)
+
 	defer func() {
 		if t.Failed() {
 			t.Log("see the --rewrite flag to re-run the benchmarks and adjust the expectations")
 		}
 	}()
 
-	for _, b := range benchmarks {
-		t.Run(b, func(t *testing.T) {
-			flags := []string{
-				"--test.run=^$",
-				"--test.bench=" + b,
-				"--test.benchtime=1x",
-			}
-			if testing.Verbose() {
-				flags = append(flags, "--test.v")
-			}
-			results := runBenchmarks(t, flags...)
-
-			for _, r := range results {
-				exp, ok := expectations.find(r.name)
-				if !ok {
-					t.Logf("no expectation for benchmark %s, got %d", r.name, r.result)
-					continue
-				}
-				if !exp.matches(r.result) {
-					t.Errorf("fail: expected %s to perform KV lookups in [%d, %d], got %d",
-						r.name, exp.min, exp.max, r.result)
-				} else {
-					t.Logf("success: expected %s to perform KV lookups in [%d, %d], got %d",
-						r.name, exp.min, exp.max, r.result)
+	var results resultSet
+	var wg sync.WaitGroup
+	concurrency := system.NumCPU() * 4 // arbitrary
+	limiter := quotapool.NewIntPool("rttanalysis", uint64(concurrency))
+	isRewrite := *rewriteFlag
+	for b, cases := range r.r {
+		quota, err := limiter.Acquire(context.Background(), 1)
+		require.NoError(t, err)
+		wg.Add(1)
+		go func(b string, cases []RoundTripBenchTestCase) {
```

Comment:
ya reckon this will be ok in parallel if multiregion tests are starting 9 servers each in CI?